afs_ioc_migration/transfer.py

```diff
@@ -0,0 +1,162 @@
+            raise
+    else:
+        raise RepoExistsError(
+            f"Repo {info.github_url} exists and has commits, aborting."
+        )
+
+    # Create the blank repo if needed
+    if repo_exists:
+        logger.info("Repo already exists, skipping creation.")
+    elif dry_run:
+        logger.info("Dry run: skipping repository creation.")
+    else:
+        logger.info(f"Creating repository at {info.github_url}")
+        gh.repos.create_in_org(
+            org=ORG,
+            name=info.name,
+            visibility="internal",
+            custom_properties={
+                "type": "EPICS IOC",
+                "protect_default": "true",
+                "protect_master": "true",
+                "protect_gh_pages": "false",
+                "required_checks": "None",
+            },
+        )
+
+    # Set repo topics
+    if dry_run:
+        logger.info("Dry run: Skip setting standard repo topics")
+    else:
+        logger.info("Setting standard repo topics")
+        gh.repos.replace_all_topics(
+            owner=ORG,
+            repo=info.name,
+            names=[
+                "epics",
+                "epics-ioc",
+                f"ecs-epics-ioc-{info.area}",
+            ],
+        )
+
+    with TemporaryDirectory() as path:
+        if dry_run:
+            top_path = Path("dry_run_afs_ioc_migration").resolve()
+            top_path.mkdir(exist_ok=True)
+            path = top_path / info.name
+            path.mkdir(exist_ok=False)
+            logger.info(f"Dry run: create repo in cwd at {path}")
+        # Clone from afs to a temporary directory
+        logger.info(f"Cloning HEAD from {afs_path} to {path} as master")
+        repo = Repo.init(path=path, mkdir=False)
+        afs_remote = repo.create_remote(name="afs_remote", url=afs_path)
+        fetch_info = afs_remote.fetch(
+            ["*:refs/remotes/afs_remote/*", "refs/tags/*:refs/tags/*"]
+        )
+        logger.info("Checking out HEAD as master")
+        afs_head = repo.create_head("master", afs_remote.refs.HEAD)
+        afs_head.checkout()
+
+        # At this point, we have all branches and tags fetched.
+        # The working directory is currently even with afs's head
+        # The head is now named "master" locally,
+        # regardless of whichever strange name it may have upstream.
+
+        # Make and commit systemic modifications (.gitignore, license, others)
+        logger.info("Adding license file")
+        license = add_license_file(cloned_path=path)
+        commit(repo, license, "MAINT: adding standard license file")
+        logger.info("Adding gitignore")
+        gitignore = add_gitignore(cloned_path=path)
+        commit(repo, gitignore, "MAINT: adding standard gitignore")
+        logger.info("Adding github templates")
+        github_templates = add_github_folder(cloned_path=path)
+        commit(repo, github_templates, "MAINT: add github templates")
+        logger.info("Updating readme")
```

Comment:
TODO from slack: look into ways to incorporate the contents of a `RELEASE_NOTES` file and of the per-tag commit notes into formal github releases

Reply:
Would we be doing anything with these formal releases?  If not, this task might rank at the bottom of the todo-priority-list for me.  

In the future of course we'd like to build CI/CD around these releases but we can surely revisit them after we migrate the repo

Reply:
Maybe my first question is better phrased as "do we have any machinery that uses the formal releases right now"

Reply:
We don't have any machinery that uses the formal releases right now. I figured this was the most correct place to record and find the original contents of `RELEASE_NOTES` (and of the per-tag commit comments) for human consumption.

Reply:
I agree this is the lowest-priority task here

Reply:
I did a little bit on this but I'm abandoning the idea because:

- low prio
- couldn't get it working quickly
- inconsistent layouts in RELEASE_NOTES files

Reply:
If we ever become concerned with api rate limits, we might spend some more time figuring out how to recover from mid-run errors.  Right now we'll probably just have to delete the repo and re-run the migration, but that's not a huge deal.